internal/users/tempentries/users.go

```diff
@@ -2,147 +2,96 @@ package tempentries
 
 import (
 	"context"
-	"crypto/rand"
-	"fmt"
 	"sync"
 
-	"github.com/ubuntu/authd/internal/users/db"
-	"github.com/ubuntu/authd/internal/users/localentries"
-	"github.com/ubuntu/authd/internal/users/types"
 	"github.com/ubuntu/authd/log"
 )
 
-type userRecord struct {
-	name  string
-	uid   uint32
-	gecos string
+type referencedUserName struct {
+	refCount uint64
+	uid      uint32
 }
 
-type temporaryUserRecords struct {
-	idGenerator IDGenerator
-	registerMu  sync.Mutex
-	rwMu        sync.RWMutex
-	users       map[uint32]userRecord
-	uidByName   map[string]uint32
+type idTracker struct {
+	mu        sync.Mutex
+	ids       map[uint32]struct{}
+	userNames map[string]*referencedUserName
 }
```

Comment:
```
commit 67f6ff8aa5483858eca275e4f2926f2cc9aa7cb6
Author: Marco Trevisan (Treviño) <mail@3v1n0.net>
Date:   Wed Jun 18 05:50:59 2025 +0200

    users/tempentries/users: Remove temporary user records
    
    [...]
    
    However, we still may want to avoid races with the pre-auth users IDs,
    since they may be registered concurrently while we're still in the process of
    registering an user or a group, so let's use a simpler and shared ID tracker
    to ensure we don't duplicate IDs across multiple elements
```

Can't we avoid these races simply by holding the userdb lock?

Reply:
the ID tracking is only used when choosing a UID for a new user. now that we have the userdb lock, we can simplify that process: there is no race anymore with other tools (assuming they also acquire a userdb lock), so we can just get the list of existing IDs once before and then generate an ID which is not in that list of existing IDs. there is no need to repeatedly generate an ID, track it and check for uniqueness via NSS. do you agree?